pydantic/_internal/_decorators.py

```diff
@@ -146,7 +152,46 @@ def __get__(
             return self.wrapped  # type: ignore[return-value]
 
 
-DecoratorInfoType = TypeVar('DecoratorInfoType', bound=DecoratorInfo)
+@slots_dataclass
+class ComputedFieldInfo:
+    """
+    A container for data from `@computed_field` so that we can access it
+    while building the pydantic-core schema.
+    """
+
+    wrapped_property: property
+    json_return_type: core_schema.JsonReturnTypes | None
+    alias: str | None
+    title: str | None
+    description: str | None
+    repr: bool
+
+    def __get__(self, obj: object | None, obj_type: type[object] | None = None) -> Callable[[], Any]:
+        self.wrapped_property = p = self.wrapped_property.__get__(obj, obj_type)
+        return p
+
+    @property
+    def setter(self) -> Callable[[Callable[[Any], None]], ComputedFieldInfo]:
+        def setter_wrapper(func: Callable[[Any], None]) -> ComputedFieldInfo:
+            self.wrapped_property = self.wrapped_property.setter(func)
+            return self
+
+        return setter_wrapper
+
+    @property
+    def deleter(self) -> Callable[[Callable[[Any], None]], ComputedFieldInfo]:
+        def deleter_wrapper(func: Callable[[Any], None]) -> ComputedFieldInfo:
+            self.wrapped_property = self.wrapped_property.deleter(func)
+            return self
+
+        return deleter_wrapper
```

Comment:
I'm having a hard time understanding what you are referring to in the linked pull request — could you give me a line to search for or similar? I'd at least be interested to understand what was originally done, even if it doesn't make sense to support here

Reply:
search for `__getattr__` and `update_g`.

Reply:
@samuelcolvin While I see the use case for that, I think it's probably a good idea not to implement that until we get specific feature requests for it (assuming we haven't already), not least of which because it will be a major pain to document (i.e., how it behaves if you _do_ call a custom method from the wrapped descriptor).

I also feel like once we have this merged, a PR for that functionality would be a lot easier to review..

Reply:
Also, I think users can potentially achieve the same functionality by doing a delayed decoration with the computed field. Namely, I think it might work to do:
```python
    class User(BaseModel):
        first: str
        last: str

        @MyCustomProperty
        def fullname(self) -> str:
            return f'{self.first} {self.last}'

        @fullname.update_g
        def new_fullname(self) -> str:
            return f'The REAL {self.first} {self.last}'

        fullname = computed_property(fullname)
```

Reply:
makes sense, let's leave it for now.

Reply:
Yup I probably went a bit far 😅